### ISSUE TITLE
1c. Add "Add email" occ call + store

### ISSUE THREAD

**Issue by LTiger14** [SAP/cloud-commerce-spartacus-storefront#1179]
We merge into **epic/guestchecout-fr1** _not develop_

To begin the guest checkout process, a user provides an email. This email is added to the cart. 

Request is: PUT /{baseSiteId}/users/anonymous/carts/{cartId - gui }/email
=> This must include the client token in the header

The aim of this ticket is to:
- [x] Link the `Continue` button in guest checkout form to perform the add email call, it should also fetch the update cart with `fields=user` (see note)
- [x] create the occ call to add email to cart
- [x] Update store with Action to add email to cart and Effect to listen for the action
- [x] Update Facade to make add email Action and *get user from cart*

***Note** A guest user will have a user in the cart like follows, if it's the case, the guest checkout guard should let the customer proceed to checkout. In order to get this you have to add `fields=user` to the cart request.

`user {
   "name": "guest",
   "uid": "472099c7-ade1-44ab-8e62-33ef66d6bd9a|xxxxxx@acme.com"
}`

When the email is added to the cart a guest user is created in the backend.

### Acceptance criteria
- Service to add email to cart
- The userId should be kept in the state

**Comment by Xymmer** [comment]
is the guid what we need here? i remember asking Kinga the same question

**Comment by mirkosoft** [comment]
Moving this back to TODO as kind of blocked with that last thing:
![image](https://user-images.githubusercontent.com/788092/55228936-d5701d80-521b-11e9-877b-c34c4abe10eb.png)
![image](https://user-images.githubusercontent.com/788092/55228955-df921c00-521b-11e9-88c9-a8f5493a4509.png)

**Comment by mirkosoft** [comment]
Moving this back to TODO as blocked by blank pages issue on the only one server supporting `user` object in the response (dev 19).

**Comment by Xymmer** [comment]
@mirkosoft blocked by which issue?

**Comment by LTiger14** [comment]
Dev-19 is the only server recent enough to have the update allowing the cart to return the user. The problem is it can't be used because of a cms problem. It needs to be updated. Gil was waiting for his server update method to be using the CSP. Since he has been busy with C3PO the server is still out of commission.

**Comment by LTiger14** [comment]
Unblocked as servers are now updated

**Comment by LTiger14** [comment]
Split this one with #3893 to make is smaller

**Comment by WeizhengSap** [comment]
My PR is here: https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/3929

**Comment by LTiger14** [comment]
QA Steps:
- Login 
- Add an item to cart
- Checkout
- Logout
- Add an item to cart
- Click proceed to checkout -> Guest login
- Enter email & confirm email (validation not tested here)
- Submit
- You should go to address form

**Comment by giancorderoortiz** [comment]
Test successful
![Screen Shot 2019-08-11 at 3 48 23 PM](https://user-images.githubusercontent.com/46171897/62838856-b8c70600-bc4f-11e9-8272-316511097df2.png)
